internal/provider/azure/credentials.go

```diff
@@ -189,6 +198,13 @@ func (c environProviderCredentials) FinalizeCredential(
 			return nil, errors.Errorf("unknown Azure cloud name %q", args.CloudName)
 		}
 
+		stderr := ctx.GetStderr()
+		fmt.Fprintln(stderr, "Note: your user account needs to have a role assignment to the")
+		fmt.Fprintln(stderr, "Azure Key Vault application (cfa8b339-82a2-471a-a3c9-0fc0be7a4093).")
```

Comment:
Was curious about this magic string here and looking back you [mentioned this method would probably go](https://github.com/juju/juju/pull/17995#discussion_r1741469185) in 4.0. Should it be included in this merge up?

Reply:
This isn't the thing that's probably going away in 4.0. The UUID here is for the Azure Key Vault app and is not what's referred to in that link. 
The thing that's going away is the handling of old credentials which need to refer to the object id of the Juju AD app.